src/lib/CollisionPrevention/CollisionPrevention.cpp

```diff
@@ -105,21 +115,88 @@ void CollisionPrevention::publish_constraints(const Vector2f &original_setpoint,
 	}
 }
 
+obstacle_distance_s CollisionPrevention::update_distance_sensor()
+{
+	obstacle_distance_s obstacle_distance = {};
+
+	for (unsigned i = 0; i < ORB_MULTI_MAX_INSTANCES; i++) {
+		const distance_sensor_s &distance_sensor = _sub_distance_sensor[i]->get();
+
+		// consider only instaces with updated data and orientations useful for collision prevention
+		if ((hrt_elapsed_time(&distance_sensor.timestamp) < RANGE_STREAM_TIMEOUT_US) && (
+			    (distance_sensor.orientation == distance_sensor_s::ROTATION_FORWARD_FACING) ||
+			    (distance_sensor.orientation == distance_sensor_s::ROTATION_RIGHT_FACING) ||
+			    (distance_sensor.orientation == distance_sensor_s::ROTATION_LEFT_FACING) ||
+			    (distance_sensor.orientation == distance_sensor_s::ROTATION_BACKWARD_FACING))) {
+
+			obstacle_distance.timestamp = distance_sensor.timestamp;
+			obstacle_distance.max_distance = distance_sensor.max_distance * 100.0f; // convert to cm
+			obstacle_distance.min_distance = distance_sensor.min_distance * 100.0f; // convert to cm
+			obstacle_distance.increment = 5;
```

Comment:
Is that hardcoded in the mavlink message that all distance sensor messages have an angle increment of 5?

Reply:
no, that information is missing. For laser I assume the FOV is 1deg so the minimum increment (5deg) is fine. For sonar, as I have written in the description, we would need to take into account the cone width. It would be nice to add an extension to the mavlink message.

Reply:
we can add the field to the uORB message and set it in the driver. What would be missing is the simulation and integration in QGC.

Reply:
Ok that sounds reasonable. This way, it would not be blocking here. But I think we should still get started with the mavlink stuff, as in the end also SITL need to work so we can properly test it

Reply:
yes, sorry. That's what I meant to say

Reply:
> For laser I assume the FOV is 1deg so the minimum increment (5deg) is fine.

Reply:
I am going to change this as I have written in the description